src/bgw/job.c

```diff
@@ -738,7 +738,7 @@ ts_bgw_job_insert_relation(Name application_name, Name job_type, Interval *sched
 	CatalogSecurityContext sec_ctx;
 	bool nulls[Natts_bgw_job] = { false };
 
-	rel = heap_open(catalog_get_table_id(catalog, BGW_JOB), RowExclusiveLock);
+	rel = relation_open(catalog_get_table_id(catalog, BGW_JOB), RowExclusiveLock);
```

Comment:
I assume the PG12 gets rid of `heap_open` and that's why we're making this change? Otherwise `heap_open` provided some protections against accidentally opening an index instead of a heap/table.

I would also have expected changing a lot of `heap_close` to `relation_close`, but I don't see that.

Reply:
yup, the `heap_*` functions were removed. I'm also surprised I haven't seen errors on `heap_close` considering it's also removed, probably, just haven't gotten it to compile enough yet.

Reply:
Why not use table_open (and index_open for indices) instead? That still contains the check

Reply:
Ah I see that has been changed in the latest commit already